File: py/pe7.py
#! /usr/bin/env python3

from sys import argv
from pe import get_prime

def pe7(n=10001):
    """
    Find the 10001st prime.
    """
    return(get_prime(n))

def main():
    argc = len(argv)
    if argc > 1:
        try:
            n = int(argv[1])
        except ValueError:
            print("  python pe7.py [integer]")
            return
    else:
        n = 10001

    print(pe7(n))

if __name__ == "__main__":
    main()

File: cpp/cli/src/pe7.cc
#include "../include/pe7.h"

#include <cstdlib>
#include <iostream>

int pe7(int argc, char **argv)
{
  using namespace std;

  uint32_t n = 10001;
  if (argc > 1) {
    char *end;
    n = strtoul(argv[1], &end, 10);
  }

  cout << "n = " << n << endl;
  cout << pe::pe7(n) << endl;

  return 0;
}
